src/cuga/backend/cuga_graph/nodes/cuga_lite/cuga_lite_graph.py

```
@@ -1148,27 +1235,78 @@ async def prepare_tools_and_apps(
 
+                memory_query = state.sub_task or get_latest_memory_query(state.chat_messages)
+                current_user_id = str(getattr(state, "user_id", "") or "").strip()
+                if current_user_id and memory_query:
+                    current_agent_id = str(configurable.get("agent_id") or "").strip()
+                    if not current_agent_id:
+                        try:
+                            from cuga.backend.server.main import app as _app
+
+                            _app_state = getattr(_app.state, "app_state", None)
+                            current_agent_id = str(getattr(_app_state, "agent_id", "") or "").strip()
+                        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)